artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyConnection.java

```diff
@@ -62,7 +61,7 @@ public class NettyConnection implements Connection {
     * here for when the connection (or Netty Channel) becomes available again.
     */
    private final List<ReadyListener> readyListeners = new ArrayList<>();
-   private final FastThreadLocal<ArrayList<ReadyListener>> localListenersPool = new FastThreadLocal<>();
+   private static final ThreadLocal<ArrayList<ReadyListener>> readyListenersPool = new ThreadLocal<>();
```

Comment:
since we are now fixing whatever was broken with ThreadLocals before. I don't think we need the FastThreadLocal any more.

Reply:
Not really... in previous versions this was just using ThreadLocal. (2.16.0 is using a regular ThreadLocal).

in more recent versions we started using the FastThreadLocal, that's just a ThreadLocal with its own Map away. (there will be a single HashMap entry on the Thread.threadLocals.

this is still an issue.. it should be static.


I could still use the FastThreadlocal here. but I don't see a reason for.

Reply:
@gemmellr even if I used FastThreadLocal here, it should still be a static declaration.

Reply:
I was commenting on your comment around FastThreadLocal...not anything about the separate change to static. I expect the change to use FastThreadLocal (in https://github.com/apache/activemq-artemis/commit/6126d926ddcdf34b740603e506878059ea663680) had nothing to do with "whatever was broken with ThreadLocals before" but was simply because its optimised for use by Netty threads and the expectation was a Netty thread will likely be calling it.

Reply:
@gemmellr I don't see a real advantage really... I will just make it static though.